lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnDriverRuntimeRestartManager.java

```diff
@@ -138,6 +142,8 @@ private synchronized void initializeListOfPreviousContainers() {
       if (this.previousContainers == null) {
         this.previousContainers = new HashSet<>();
       }
+
+      yarnContainerManager.onContainersRecovered(Collections.unmodifiableSet(this.previousContainers));
```

Comment:
I like the immutability here. However, wouldn't it make sense to make `this.previousContainers` immutable?